src/io/readers/tdf_utils.rs

```diff
@@ -0,0 +1,124 @@
+use crate::io::readers::file_readers::sql_reader::frame_groups::SqlWindowGroup;
+use crate::ms_data::QuadrupoleSettings;
+
+type SpanStep = (usize, usize);
+
+enum QuadWindowExpansionStrategy {
+    None,
+    Even(usize),
+    Uniform(SpanStep),
+}
+
+fn scan_range_subsplit(
+    start: usize,
+    end: usize,
+    strategy: &QuadWindowExpansionStrategy,
+) -> Vec<(usize, usize)> {
+    let out = match strategy {
+        QuadWindowExpansionStrategy::None => {
+            vec![(start, end)]
+        },
+        QuadWindowExpansionStrategy::Even(num_splits) => {
+            let sub_subwindow_width = (end - start) / (num_splits + 1);
+            let mut out = Vec::new();
+            for sub_subwindow in 0..num_splits.clone() {
+                let sub_subwindow_scan_start =
+                    start + (sub_subwindow_width * sub_subwindow);
+                let sub_subwindow_scan_end =
+                    start + (sub_subwindow_width * (sub_subwindow + 2));
+
+                out.push((sub_subwindow_scan_start, sub_subwindow_scan_end))
+            }
+            out
+        },
+        QuadWindowExpansionStrategy::Uniform((span, step)) => {
+            let mut curr_start = start.clone();
+            let mut curr_end = start + span;
+            let mut out = Vec::new();
+            while curr_end < end {
+                out.push((curr_start, curr_end));
+                curr_start += step;
+                curr_end += step;
+            }
+            out
+        },
+    };
+    out
+}
+
+pub fn expand_quadrupole_settings(
+    window_groups: &[SqlWindowGroup],
+    quadrupole_settings: &[QuadrupoleSettings],
+) -> Vec<QuadrupoleSettings> {
+    // Read the 'NUM_SUB_SUB_SPLITS' from env variables ... default to 1
+    // (for now)
+
+    let splits = match std::env::var("NUM_SUB_SUB_SPLITS") {
```

Comment:
Obviously this would be passed as an argument and not read from the env. variables. IMO